fizzbuzz/src/main/java/fizz.buzz/number/factory/AbstractNumberFactory.java

```diff
@@ -0,0 +1,37 @@
+package fizz.buzz.number.factory;
+
+import fizz.buzz.number.Number;
+import fizz.buzz.number.NumberFactory;
+
+import java.lang.reflect.InvocationTargetException;
+
+public abstract class AbstractNumberFactory implements NumberProcessor, NumberFactory {
+
+    protected NumberFactory successor;
+
+    @Override
+    public void setSuccessor(NumberFactory nextFactory) {
+        successor = nextFactory;
+    }
+
+    @Override
+    public Number build(Integer input) {
+        if (input % getDivisor() == 0) {
+            return instantiateNumber(input);
+        }
+
+        return successor.build(input);
```

Comment:
True. I've been thinking about the same. The downside I see for using the constructor would be harming the readability of `NumberFactoryImpl#chainFactories`. Right now, we can easily read how the factories are chained via the calls to `setSuccessor`, I wonder if we changing to constructor injection wouldn't harm readability. What do you think?